public/sw.js

```diff
@@ -165,66 +193,147 @@ async function fetchThenCache(request) {
+          return response;
+        });
+      });
       // If there's no response from cache, we console log that the request failed
       if (response) {
         return response;
-      } else if (showDeveloperConsoleLog) {
+      } else if (showDeveloperConsoleLog)
         console.log(`%c${errorEmoji} Failed to get ${request.url} from cache`, errorLog);
-      }
     });
 }
 
 /**
- * Caches all of the static files that are listed in the array staticCache
+ * Caches all of the static files
  *
- * If all files are cached successfuly, its success is console logged
- * Else, console log that caching all files failed
- *
- *  @return {Promise} A promise with the result of caching the static files
+ * The files cached are the static files' links in the staticCache array and the Guest view data.
+ * This data includes both events and involvements for the current session. If all files are cached
+ * successfuly, its success is console logged. Vice versa if it fails.
  */
 async function cacheStaticFiles() {
-  return await caches.open(cacheVersion).then(cache => {
-    cache
-      .addAll(staticCache)
-      .then(() => {
-        console.log(`%c${successfulEmoji} Cached Static Files Successfully`, successfulLog);
+  // Determines if all files were successfully cached
+  let cachedSuccessfully = true;
+
+  // CACHES LOCAL FILES
+  await caches.open(cacheVersion).then(cache => {
+    cache.addAll(staticCache).catch(error => {
+      cachedSuccessfully = false;
+      if (showDeveloperConsoleLog) console.log(`%c${errorEmoji} Error: ${error.message}`, errorLog);
+    });
+  });
+
+  // GETS THE CURRENT SESSION
+  await fetch((request = new Request(`${apiSource}/api/sessions/current`)))
+    .then(async fetchResponse => {
+      // Checks to make sure the response of the fetch is okay
+      if (fetchResponse.statusText === 'OK') {
+        // Adds fetch response to cache
+        await caches.open(cacheVersion).then(cache => {
+          cache.put(request.url, fetchResponse.clone());
+        });
+
+        return fetchResponse.json();
+      } else {
+        cachedSuccessfully = false;
+        if (showDeveloperConsoleLog) {
+          console.log(
+            `%c${warningEmoji} Bad Response: Status - ${fetchResponse.status} \n\t\tURL: ${fetchResponse.url}`,
+            warningLog,
+          );
+        }
+      }
+    })
+    .then(session => {
+      currentSession = session;
+      dynamicCacheTwo.push(`${apiSource}/api/activities/session/${session.SessionCode}`);
+      dynamicCacheTwo.push(`${apiSource}/api/activities/session/${session.SessionCode}/types`);
+    })
+    // Catches any errors from the fetch
+    .catch(error => {
+      cachedSuccessfully = false;
+      if (showDeveloperConsoleLog)
+        console.log(`%c${errorEmoji} Error: ${error.message} \n\t\tURL: ${request.url}`, errorLog);
+    });
+
+  // RETRIEVES All DATA
+  dynamicCacheTwo.forEach(async item => {
+    await fetch((request = new Request(item)))
```

Comment:
It does a fetch for 5 different URLs. These fetches are to get the data that's needed to view the Involvements Tab (only the current session) and the Events Tab as guests offline. This won't necessarily be a performance issue since each fetch is asynchronous. I do plan on refactoring and breaking up the service worker to multiple parts because it's very cluttered.